contracts/margin/external/ERC20/ERC20Long.sol

```diff
@@ -99,31 +92,37 @@ contract ERC20Long is ERC20Position {
             uint256 /* allowedCloseAmount */
         )
     {
+        uint256 positionBalance = Margin(DYDX_MARGIN).getPositionBalance(POSITION_ID);
+
         uint256 requestedTokenAmount = MathHelpers.getPartialAmount(
             requestedCloseAmount,
             positionPrincipal,
-            totalSupply_
+            positionBalance
         );
 
+        // if user has enough tokens, allow the close to occur
         if (requestedTokenAmount <= balance) {
             return (requestedTokenAmount, requestedCloseAmount);
         }
 
         // The maximum amount of principal able to be closed without using more heldTokens
         // than balance
-        uint256 maxAllowedCloseAmount = MathHelpers.getPartialAmount(
+        uint256 allowedCloseAmount = MathHelpers.getPartialAmount(
             balance,
-            totalSupply_,
+            positionBalance,
             positionPrincipal
         );
 
-        uint256 tokenAmount = MathHelpers.getPartialAmount(
-            maxAllowedCloseAmount,
+        // the new close amount should not be higher than what was requested
+        assert(allowedCloseAmount < requestedCloseAmount);
```

Comment:
should this be `<=`?

Reply:
If it were equal, then it would have returned up above